src/python/pants/backend/native/subsystems/binaries/gcc.py

```diff
@@ -43,6 +43,7 @@ def _common_lib_dirs(self, platform):
     lib64_tuples = platform.resolve_for_enum_variant({
       'darwin': [],
       'linux': [('lib64',)],
+      'none': [],
```

Comment:
Hm... this is strange. When would this end up being used?

It's possible that having a better name than "none" ("cross-platform"? "unknown"?) would help, but in general, I don't understand the semantics of having a separate set of "cross platform" args.

Reply:
yeah this is an unfortunate side effect of needing values for every enum variant in this call. I never expect platform to be none in the native subsystem because it always uses `Platform.current` to resolve that platform. I could modify `resolve_for_enum_variant` to produce a default of None instead of raising. I made this change because I want platform to have a 'platform.none' variant so we can upcast EPR's to MPEPRs on the python side and keep the rust intrinsics simpler.

Reply:
> I made this change because I want platform to have a 'platform.none' variant so we can upcast EPR's to MPEPRs on the python side and keep the rust intrinsics simpler.

But what does "Platform.none" actually mean? Is it supposed to mean: "it's cross platform"? If so, can we rename it?

Reply:
Yea I think I will make a new enum called PlatformConstraint that has a PlatformLiteral and a cross platform type. Instead of using this type for both, which is clearly hacky.